src/main/java/org/ambraproject/wombat/model/RelatedArticleType.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2019 Public Library of Science
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package org.ambraproject.wombat.model;
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableSet;
+
+@AutoValue
+public abstract class RelatedArticleType implements Comparable<RelatedArticleType> {
+  public abstract int getSortOrder();
+  public abstract String getDisplayName();
+  public abstract String getName();
+  public boolean isPertainsToRelation() {
```

Comment:
The names `isPertainsToRelation` and `isHasRelation` are really hard for me to parse.  The comment helps, though I think the inverse nature of the two methods would be clearer if the definition of `isPertainsToRelation` was directly below the definition of `isHasRelation`.

Maybe an `isInverted` method could replace it all.

Reply:
It's basically a helper just for the freemarker template. I can certainly move it closer. 

What do you mean by an `isInverted` method?

Reply:
Thanks.  Regarding `isInverted`, I was just trying to come up with something less awkward than `isHas...` but maybe it's not worth refactoring, as they are commented, and as helpers they have the virtue, for now at least, of being named similarly to the text in the UI.